src/main/java/hudson/plugins/ec2/EC2SlaveMonitor.java

```diff
@@ -51,8 +51,12 @@ private void removeDeadNodes() {
                         ec2Slave.terminate();
                     }
                 } catch (AmazonClientException e) {
-                    LOGGER.info("EC2 instance is dead and failed to terminate: " + ec2Slave.getInstanceId());
-                    removeNode(ec2Slave);
+                    if (new RequestExpiredPredicate().test(e)) {
```

Comment:
Maybe this is better as a private static instance

Reply:
@res0nance Thanks for taking a look. I've updated the source branch with the code we are using internally. I'm not crazy about a static instance either, but it's better than creating the instance every check. I do see I'm doing the same thing in the CloudHelper class. I'll move that to a static field as well for now.